### PR TITLE
Components become optional

### LINKED ISSUE #135: Components move from Required to Optional
Release 5 introduced “Components” as subunits of pieces of content - for example, a dataset may be a component of a journal article. Feedback from repositories and publishers suggests that few users actively wish to analyse usage at this level of detail, while the requirement to deliver such granular information makes generating an Item Report very difficult. In R5.1 we are making Components optional - that is, allowing report providers to deny a request for Components by delivering an Exception.

Affects
- Section 3.2.2 Report Body, specifically the subsection on Component Item Description and Identifiers
- Section 3.3.8 Report Filters and Report Attributes, table 3.x
- Section 4.4.2 Column Headings/Elements and 4.4.3 Filters and Attributes, both for Item Reports
- Appendix A, Glossary of Terms
- Appendix F, Handling Errors and Exceptions

### PR DESCRIPTION
**Pull request by tashalouizamc**
Specifying that provision of component usage reports is at the discretion of report providers

### CODE REVIEW

File: source/03-specifications/03-counter-report-common-attributes-and-elements.rst [hunk 1]
```diff
@@ -703,11 +703,11 @@ Table 3.x (below): Report Attributes
        (JSON)
 
    * - Include_Component_Details
-     - Specifies whether to include the component columns/elements (see Table 3.k) in the report. Permissible values are False (default) and True.
+     - Specifies whether to include the component columns/elements (see Table 3.k) in the report, where report providers offer component usage reporting. Permissible values are False (default) and True.
      - IR
 
    * - Include_Parent_Details
-     - Specifies whether to include the parent columns/elements (see Table 3.j) in the report. Permissible values are False (default) and True.
+     - Specifies whether to include the parent columns/elements (see Table 3.j) in the report, where report providers offer component usage reporting. Permissible values are False (default) and True.
```
Comment (berndoberknapp) [inline source/03-specifications/03-counter-report-common-attributes-and-elements.rst]: Parents are not related to components, so this change must be reverted.

File: source/04-reports/04-item-reports.rst [hunk 2]
```diff
@@ -147,7 +147,7 @@ Table 4.o (below): Header for Item Report and Standard Views of the Item Report
...
+The following elements MUST appear in the tabular report in the order they appear in the table below. For guidance on how these elements appear in the JSON format, refer to the COUNTER_SUSHI API Specification (see :numref:`sushi` below). Mandatory (M) elements MUST be included in the report. For report providers who elect to offer component usage information, the Parent and Component elements MUST only be included in the COUNTER Report if called for (C) via Include_Parent_Details and Include_Component_Details, respectively (they are not supposed to be selected individually). If they are included, then the corresponding Include_Parent_Details=True or Include_Component_Details=True MUST be included in the Report_Attributes header. The other elements also MUST only be included if called for (C), and if included they MUST be listed in Attributes_To_Show in the Report_Attributes header.
```
Comment (berndoberknapp) [inline source/04-reports/04-item-reports.rst]: Parent reporting is required. I would suggest to split this sentence into two sentences, one for parent and one for component reporting.

File: source/appendices/f-handling-errors-and-exceptions.rst [hunk 3]
```diff
@@ -253,6 +253,15 @@ Table F.1 (below): Exceptions
 
        Note: The server is expected to ignore unsupported report attributes and continue to process the request, returning data that is available without the report attribute being applied.
 
+   * - Components Not Supported
+     - Warning\ |br|\ |lb|
+       Error
+     - 3063
+     - 200
+     - The request contained Component attribute values that are not supported by the server. The server should list the names of unsupported report attribute values in the additional Data element.
```
Comment (berndoberknapp) [inline source/appendices/f-handling-errors-and-exceptions.rst]: The component elements cannot be requested individually. Suggestion: The request contained include_component_details=True, but reporting on component usage is not supported.

**Review by berndoberknapp — CHANGES_REQUESTED** [review]
Please see the inline comments.

**Comment by tashalouizamc** [comment]
Comments addressed in new commit 2023-03-15

**Review by berndoberknapp — APPROVED** [review]

**Pull request merged**